pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -629,13 +673,72 @@ func (b *OFBridge) RetryInterval() time.Duration {
 	return b.retryInterval
 }
 
+func (b *OFBridge) queryTableFeatures() {
+	mpartRequest := &openflow13.MultipartRequest{
+		Header: openflow13.NewOfp13Header(),
+		Type:   openflow13.MultipartType_TableFeatures,
+		Flags:  0,
+	}
+	mpartRequest.Header.Type = openflow13.Type_MultiPartRequest
+	mpartRequest.Header.Length = mpartRequest.Len()
+	tableFeatureChan := make(chan *openflow13.MultipartReply)
+	b.mpartRplChs[mpartRequest.Header.Xid] = tableFeatureChan
+	go b.processTableFeatures(mpartRequest.Header.Xid, tableFeatureChan)
+	b.ofSwitch.Send(mpartRequest)
+}
+
+func (b *OFBridge) processTableFeatures(xid uint32, ch chan *openflow13.MultipartReply) {
+	header := openflow13.NewOfp13Header()
+	header.Type = openflow13.Type_MultiPartRequest
+	requests := make([]*openflow13.MultipartRequest, 0)
+	tableCount := 0
+	for tableCount < int(LastTableID-1) {
+		select {
+		case rpl := <-ch:
+			request := &openflow13.MultipartRequest{
+				Header: header,
+				Type:   openflow13.MultipartType_TableFeatures,
+				Flags:  1,
+			}
+			for _, body := range rpl.Body {
+				request.Body = append(request.Body, body)
+			}
+			tableCount += len(rpl.Body)
+			request.Length = request.Len()
+			requests = append(requests, request)
+		}
+	}
+	close(ch)
+	delete(b.mpartRplChs, xid)
+	for i := range requests {
+		req := requests[i]
+		for _, body := range req.Body {
+			tableFeature := body.(*openflow13.OFPTableFeatures)
+			// Modify table name if the table is in the pipeline, otherwise use the default table features.
+			// OVS doesn't allow to skip any table except the hidden table (always the last table) in a table_features
+			// request. So use the table features which is got from the initial query result for the tables that Antrea
+			// doesn't define in the pipeline.
+			if t, ok := b.tableCache[tableFeature.TableID]; ok {
+				// Set table name with the configured value.
+				copy(tableFeature.Name[0:], t.name)
```

Comment:
Is `[0:]` needed when it starts from 0?

Reply:
tableFeature.Name is a byte array, we need to use [0:] to translate it to a slice to copy the bytes